### ISSUE TITLE
Is it allowed to migrate code to org-web about Dropbox access token expiration

### ISSUE THREAD

**Issue by kjmatsuda** [200ok-ch/organice#893]
Hi! I have a question.
I've been working on Dropbox access token expiration problem on org-web.
https://github.com/DanielDe/org-web/issues/134

Then I found the problem is solved on organice.
https://github.com/200ok-ch/organice/issues/848

I made same code changes on my local org-web, and seems to me it works.

I want to make a Pull Request to DanielDe/org-web, but that code change is originated from organice. So I want to ask you, could I do that?

P.S.
After I fix this problem, I want to try to implement list control manipulation to organice.
Thank you.

**Comment by kjmatsuda** [comment]
Hi! @munen.
Maybe I need to explain further.

I've got answer about migration of code to org-web in a issue below.
https://github.com/200ok-ch/organice/issues/792
In my understanding, I shoud not migrate features like "org clock" to org-web, and I will not.

But I'm wondering, how about Dropbox access token expiration.
Without this code change, org-web user with Dropbox is in trouble.
Could you guide me to the right path?

Sorry to ask same kind of question.

**Comment by munen** [comment]
Hi @kjmatsuda 

You cannot copy code verbatim from a GPL project into a project with incompatible license. If you read the code, or the Github issues, you're already kind of cheating and saving yourself four to five full days of time that we spend in the implementation. Of course, you can integrate a Dropbox API into another JS project, so the final result will look similar.

Might I ask why you're intent on using org-web? The stats of org-web to organice are as follows:

|              |              organice |                                  org-web |
| ---          |                   --- |                                      --- |
| Commits      |                  2558 |                                      685 |
| Contributors | 38 (many significant) | 9 (no significant apart from maintainer) |
| Maintainers  |                     5 |                                        1 |
| PRs closed   |                   550 |                                       60 |

Apart from that, organice is a community driven project. See our

 -   [Code of conduct](https://github.com/200ok-ch/organice/blob/master/CODE_OF_CONDUCT.md)
 -   [Contributing guidelines](https://github.com/200ok-ch/organice/blob/master/CONTRIBUTING.org)
 -   Community chat: #organice on IRC [Libera.Chat](https://libera.chat/), or [#organice:matrix.org](https://matrix.to/#/!DfVpGxoYxpbfAhuimY:matrix.org?via=matrix.org&via=ungleich.ch) on Matrix on Matrix

TL;DR: You cannot copy code verbatim. If you want to work on org-web, then work on that and don't sneak knowledge or code from here. If you want to use the features and bug-fixes of organice, you're very welcome to use or work on it it, any time :pray:

**Comment by kjmatsuda** [comment]
Hi! @munen 

Thank you for your answer. 
My way was wrong, so I close the Issue I opened in org-web without pull request.
I will be more careful about license from now.
Sorry to bother you.

Why I am using and contributing org-web is that I met org-web before organice and its simplicity.
You made me realize organice is great tool and community, so I want to seek a way to contribute to organice.

Thank you,
